src/abstract_variables.jl

```diff
@@ -141,6 +141,9 @@ struct InputVariable{VD<:VarDims, UT<:Units} <: AbstractVariable{VD}
     "Physical untis associated with this state variable"
     units::UT
 
+    "Interval domain on which scalar fields of this variable are defined"
+    domain::DT
```

Comment:
What's the domain used for?

Reply:
I think I added it to `InputVariable` just so the constructors would be consistent. Omitting it made some logical sense (don't really need it for inputs) but caused some issues elsewhere, though I can't remember exactly what it was.